www/content/nexusraw.md

```diff
@@ -0,0 +1,35 @@
+---
+title: Nexus Raw
+series: customization
+hideFromIndex: true
+weight: 117
+---
+
+GoReleaser supports building and pushing artifacts into Sonatype Nexus repositories of type `raw`.
+
+## How it works
+
+You can declare multiple Nexus instances.
+All binaries generated by your `builds` section will be pushed to
+each configured Nexus.
+
+If you have only one Nexus instance,
+the configuration is as easy as adding to your `.goreleaser.yml` file 
+the `URL` of your Nexus instance, the `name` of the repository,
+the `directory` where you want to upload inside that repository
+and your credentials (`username` and `password`):
+
+```yaml
+nexus_raws:
+  -
+    url: https://<Your-Instance>:<Your-Port>
+    repository: example-repo
+    directory: example-folder
+    username: your-username
+    password: your-password
```

Comment:
we should probably get the password from an environment variable?

Reply:
In order to do this, shoud I use the `EnvFiles` struct in the `config/config.go` file together with the loadEnv function?

Seems like it is very tailor-made to the git release process.

Should I load the env variable in my pipe instead?

Reply:
take a look on the artifactory pipe, it uses the internal `http` package, which maybe you can try to use here as well, which should handle secrets and etc

https://goreleaser.com/artifactory/

Reply:
The problem I see with it is that the internal http package has a hardcoded PUT method, so in order to use it I would need to refactor it to be able to use POST, which is the verb that Nexus accepts for uploads.

Reply:
I think we can refactor that first then? Maybe in another PR even?

Basically adding a `method` option which defaults to `PUT`, right? Or are there more things?

Reply:
I've been checking out the http package, to assess the refactor to include POST, and I see that PUT is highly hardcoded in it, it wouldn't be a trivial refactor.

I'm sorry to say I don't have time enough to attack this challenge at the moment. 

Would you mind if I finish this Nexus feature loading the environment variable from inside my pipe, and I can hopefully tackle the refactoring of the http package later on?

Reply:
Sorry, I rather not.

I can try to take a look at the `http` package as well...

Reply:
@antonienko would https://github.com/goreleaser/goreleaser/pull/1246 work?

Reply:
On a first quick review, it looks great. As soon as it is merged I'll try and complete this PR using it.

Reply:
merged to master :)